torch/csrc/cuda/comm.cpp

```diff
@@ -326,12 +344,22 @@ std::vector<at::Tensor> scatter(
       : tensor.chunk(/*chunks=*/devices.size(), /*dim=*/dim);
   at::cuda::OptionalCUDAStreamGuard cuda_guard;
   for (size_t i = 0; i < chunks.size(); ++i) {
-    const auto device_index = static_cast<int16_t>(devices[i]);
+    TORCH_CHECK(
+        devices[i].is_cuda(),
+        "Expected CUDA devices, but got '",
+        devices[i],
+        "'");
+    TORCH_CHECK(
+        devices[i].has_index(),
+        "Expected devices with specified indices, but got '",
+        devices[i],
+        "'");
```

Comment:
this check occurs several times, I wonder if it worth it to wrap this into a inline function?